### DISCUSSION TITLE
problemas con el lab 2.4 ejercicio 1

### DISCUSSION THREAD

**Discussion by ju4nm4n0t4s** [rramosp/ai4eng.v1#239] [02 - PYTHON]
buenas tardes termine mi código para el punto 1 pero al momento de le revisión me aparece un cero, lo que me parece extraño ya que mi función realiza la separación de las cartas y al observar los resultados con lo que entrega el código de evaluación me aparece que debe ser igual  

def make_heaps(c, n_heaps=3):
    assert n_heaps%2==1, "must have an odd number of heaps"
    assert len(c)%n_heaps==0, "the length of the deck must be a multiple of the number of heaps"
    h = (np.array(np.random.permutation(c)).reshape(int(n_heaps),int(len(c)/n_heaps))).tolist()
    return h

**Comment by rramosp** [comment]
tienes que ponernos más información ... cómo es el resultado que te arroja
el sistema de correción

On Thu, Mar 2, 2023 at 4:34 PM ju4nm4n0t4s ***@***.***> wrote:

> buenas tardes termine mi código para el punto 1 pero al momento de le
> revisión me aparece un cero, lo que me parece extraño ya que mi función
> realiza la separación de las cartas y al observar los resultados con lo que
> entrega el código de evaluación me aparece que debe ser igual
>
> def make_heaps(c, n_heaps=3):
> assert n_heaps%2==1, "must have an odd number of heaps"
> assert len(c)%n_heaps==0, "the length of the deck must be a multiple of
> the number of heaps"
> h =
> (np.array(np.random.permutation(c)).reshape(int(n_heaps),int(len(c)/n_heaps))).tolist()
> return h
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rramosp/ai4eng.v1/discussions/239>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADJRZIMFTATZZHAMLLYOJRDW2EHABANCNFSM6AAAAAAVN4ZWCU>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

-- 
*Aviso legal:* El contenido de este mensaje y los archivos adjuntos son 
confidenciales y de uso exclusivo de la Universidad Nacional de Colombia. 
Se encuentran dirigidos sólo para el uso del destinatario al cual van 
enviados. La reproducción, lectura y/o copia se encuentran prohibidas a 
cualquier persona diferente a este y puede ser ilegal. Si usted lo ha 
recibido por error, infórmenos y elimínelo de su correo. Los Datos 
Personales serán tratados conforme a la Ley 1581 de 2012 y a nuestra 
Política de Datos Personales que podrá consultar en la página web 
www.unal.edu.co <http://www.unal.edu.co/>.* *Las opiniones, informaciones, 
conclusiones y cualquier otro tipo de dato contenido en este correo 
electrónico, no relacionados con la actividad de la Universidad Nacional de 
Colombia, se entenderá como personales y de ninguna manera son avaladas por 
la Universidad.

**Comment by ju4nm4n0t4s** [comment]
![image](https://user-images.githubusercontent.com/110548663/222592280-9eb3d9e3-d390-4363-b515-13268a3b5958.png)

**Comment by Felipe-RA** [comment]
Saludos @ju4nm4n0t4s 

El problema está en que los `heaps` que estas creando son incorrectos según las instrucciones dadas (fijate que el corrector te indica un **expected output**, que es como lo deberías de tener)

Revisa nuevamente el procedimiento de los vídeos, debes distribuir las cartas del `deck` una a una en los `heaps`

Veo que intentaste generar tu respuesta en una línea.

Una forma en la que puedes hacer esto es con un `list comprehension` que recorra `range(n_heaps)` y tome una carta del mazo cada `n_heaps` 

Si deseas tomar este camino, te recomiendo leerte [este tutorial sobre el slicing](https://www.geeksforgeeks.org/python-list-slicing/)

Y este sobre [list comprehension](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)

Estamos pendientes en caso de que necesites una mano extra!

**Comment by ju4nm4n0t4s** [comment]
listo, ya entendí mi problema y como menciono antes era por la distribución muchas gracias.

def make_heaps(c, n_heaps=3):
    assert n_heaps%2==1, "must have an odd number of heaps"
    assert len(c)%n_heaps==0, "the length of the deck must be a multiple of the number of heaps"
    h = (np.transpose(c.reshape(int(len(c)/n_heaps),int(n_heaps)))).tolist()
    return h

no estoy muy familiarizado con las listas compuestas pero gracias.
